x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -172,9 +194,16 @@ const getActionDetailsList = async ({
       matchedResponses
     );
 
+    if (showHostsInfo) {
+      hosts = action.agents.reduce<Record<string, { name: string }>>((acc, id) => {
```

Comment:
Use `hosts` type of `ActionDetails`:

```suggestion
      hosts = action.agents.reduce<Required<ActionDetails>['hosts']>((acc, id) => {
```